src/gluonts/torch/scaler.py

```diff
@@ -60,21 +58,14 @@ def forward(
 
         scale = ts_sum / torch.clamp(num_observed, min=1)
 
-        # Set default_scale for time-series which are all zeros.
         # If `default_scale` is provided, we use it, otherwise we use the scale
         # of the batch.
-        # Note: We want to support tracing and to remove branching we we always
-        # calculate the batch_scale. Also, using `where` allows us to set
-        # values conditionally.
-        batch_sum = ts_sum.sum(dim=0)
-        batch_observations = torch.clamp(num_observed.sum(0), min=1)
-        batch_scale = torch.squeeze(batch_sum / batch_observations)
-
-        default_scale = torch.where(
-            self.default_scale > 0.0,
-            self.default_scale,
-            batch_scale,
-        )
+        if self.default_scale is not None:
```

Comment:
What is the effect if this change on tracing?

Reply:
I think that, as long as `self.default_scale` stays constant (which it should) then tracing should be fine with it. But I need to verify

Reply:
Also according to the warning here https://pytorch.org/docs/stable/generated/torch.jit.trace.html#torch-jit-trace it should be fine as long as the control flow is not affected by the value in input tensors (and I would add: if it’s not otherwise changed throughout the model execution, like the `forward` changing the value of `self.default_scale` here for some reason)